pyquil/paulis.py

```diff
@@ -595,6 +602,37 @@ def get_programs(self):
         return programs, coefficients
 
 
+def simplify_pauli_sum(pauli_sum):
+    # You might want to use a defaultdict(list) here, but don't because
+    # we want to do our best to preserve the order of terms.
```

Comment:
Fun fact: dictionaries are ordered by default in Python 3.6+

Reply:
This particular issue would still be a problem on python 3.5

Reply:
We can disable the py2 tests and wait til 2.0 to start introducing blatantly python 3 features